app/code/Magento/Swatches/view/base/web/js/swatch-renderer.js

```diff
@@ -951,6 +951,11 @@ define([
             );
 
             isShow = typeof result != 'undefined' && result.oldPrice.amount !== result.finalPrice.amount;
+            if (isShow) {
+                $productPrice.find('span:first').addClass('special-price');
+            } else {
+                $productPrice.find('span:first').removeClass('special-price');
+            }
```

Comment:
In my opinion you can use `toggleClass`

https://api.jquery.com/toggleclass/

Reply:
Hello @lbajsarowicz 
I have update PR as per your suggestion
can you review it?